### PR TITLE
48 proximity slider

### PR DESCRIPTION
**Pull request by galbwe**
A backend engineer's hilarious attempt at writing frontend code. I apologize in advance if my react and scss are way more verbose than they need to be, and I would appreciate any feedback on how to make it more concise and readable.

### CODE REVIEW

File: frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js [hunk 1]
```diff
@@ -1,9 +1,132 @@
...
+    this.state = {
+      position: position,
+      leftWidth: `${Math.floor(position*100)}%`,
+      rightWidth: `${Math.floor((1 - position)*100)}%`,
+      dragging: false,
+    }
+  }
+
+  handleMouseDown(e) {
+    console.log('mouse down')
```
Comment (ASTPMeetup) [inline frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js]: console log here can go.

File: frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js [hunk 2]
```diff
@@ -1,9 +1,132 @@
...
+    this.setState({
+      dragging: true,
+    })
+  }
+
+  handleMouseMove(e) {
+    if (this.state.dragging) {
+      console.log('dragging')
```
Comment (ASTPMeetup) [inline frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js]: console log here can go.

File: frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js [hunk 3]
```diff
@@ -1,9 +1,132 @@
...
+      this.setState({
+        position: position,
+        leftWidth: `${Math.floor(position*100)}%`,
+        rightWidth: `${Math.floor((1 - position)*100)}%`,
+      })
+    }
+  }
+
+  stopDragging(e) {
+    console.log('mouse up')
```
Comment (ASTPMeetup) [inline frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js]: console log here can go.

File: frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js [hunk 4]
```diff
@@ -1,9 +1,132 @@
...
+          className="proximity-slider-track-left"
+          style={{width: this.state.leftWidth}}>
+        </div>
+        <div
+          className="proximity-slider-track-thumb"
+          onMouseDown={this.handleMouseDown}
+        ></div>
+        <div
+          className="proximity-slider-track-right"
+          style={{width: this.state.rightWidth}}>
```
Comment (ASTPMeetup) [inline frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js]: style attributions can be assigned to a class

File: frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js [hunk 5]
```diff
@@ -1,9 +1,132 @@
...
+        <div
+          className="tick-spacing"
+          style={{
+            width: tickSpacingWidth,
+          }}
+        ></div>
+    )
+  };
+  const renderTickLabel = label => {
+    console.log("rendering tick label")
```
Comment (ASTPMeetup) [inline frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js]: console log here can go.

File: frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js [hunk 6]
```diff
@@ -1,9 +1,132 @@
...
+    return <div className="tick-label">{label}</div>
+  };
+  return (
+    <React.Fragment>
+    <div className="proximity-slider-ticks">
+      {labels.slice(0, -1).map(label => renderTickSpacing())}
```
Comment (ASTPMeetup) [inline frontend/src/components/panels/project_filters_panel/filter_map_components/ProximitySlider.js]: This logic can be returned by a helper;

Ex: `tickSpacing() {
return labels.slice(0, -1).map(label => renderTickSpacing())
}`

**Comment by clementkng** [comment]
@galbwe I have some time, do you mind if I clean this PR up? I also to interact w/ React and scss.

**Comment by galbwe** [comment]
@clementkng sure, go for it.

**Pull request closed without merging**